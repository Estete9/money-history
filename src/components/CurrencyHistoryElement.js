import PropTypes from 'prop-types';

function CurrencyHistoryElement({ currency }) {
  return (
    <>
      <div>{currency.year}</div>
      <div>{`${currency.currencySymbol} ${currency.value} = EUR 1`}</div>
    </>
  );
}

CurrencyHistoryElement.propTypes = {
  currency: PropTypes.shape({
    currencySymbol: PropTypes.string.isRequired,
    year: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  }),
};

CurrencyHistoryElement.defaultProps = {
  currency: {
    currencySymbol: '',
    year: 0,
    value: 0,
  },
};
export default CurrencyHistoryElement;
