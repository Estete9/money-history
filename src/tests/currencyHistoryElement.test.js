import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyHistoryElement from '../components/CurrencyHistoryElement';

test('should render CurrencyHistoryElement', () => {
  const currency = {
    year: '2023',
    currencySymbol: 'GBP',
    value: '1.65',
  };
  const component = renderer.create(<CurrencyHistoryElement currency={currency} />);
  const historyElementTree = component.toJSON();
  expect(historyElementTree).toMatchSnapshot();
});
