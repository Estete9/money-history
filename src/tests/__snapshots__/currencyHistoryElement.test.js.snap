// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrencyHistoryElement 1`] = `
<div
  className="currencyHistoryElementWrapper"
>
  <div>
    2023
  </div>
  <p
    className="currencyConversion"
  >
    GBP 1.65 = EUR 1
  </p>
  <img
    alt="enter currency history button"
    className="historyArrowImg"
    src="arrow-right.svg"
  />
</div>
`;
